etcddb/payment_channel_storage.go

```diff
@@ -74,21 +74,24 @@ type PaymentChannelStorageServerConf struct {
 // is not set in the configuration file
 func GetPaymentChannelStorageServerConf(vip *viper.Viper) (conf *PaymentChannelStorageServerConf, err error) {
 
-	// conf = &PaymentChannelStorageServerConf{Scheme: "http", ClientPort: 2379, PeerPort: 2380, Enabled: true}
-	conf = &PaymentChannelStorageServerConf{}
-	defaultConf := normalizeDefaultConf(DefaultPaymentChannelStorageServerConf)
-	err = json.Unmarshal([]byte(defaultConf), conf)
-
-	if err != nil {
-		return
+	conf = &PaymentChannelStorageServerConf{
+		ID:         "storage-1",
+		Scheme:     "http",
+		Host:       "127.0.0.1",
+		ClientPort: 2379,
+		PeerPort:   2380,
+		Token:      "unique-token",
+		Cluster:    "storage-1=http://127.0.0.1:2380",
+		Enabled:    false,
 	}
 
-	if vip.Get(PaymentChannelStorageServerKey) == nil {
+	if !vip.InConfig(PaymentChannelStorageServerKey) {
```

Comment:
It should be enough to add values into default config as you made in config/config.go . Default config is loaded as defaults to the viper and if user skipped some part of the configuration the default one will be returned. So I would propose remove manual defaults handling.

Reply:
Viper UnmarshalKey(key) does not read values from defaults.
I have filled a separate issue on this change: #74

Reply:
To make it more clear it actually reads but it suffers from issue which is the similar to https://github.com/spf13/viper/issues/559

Ok, lets merge it and continue discussion in the issue https://github.com/singnet/snet-daemon/issues/74